llm-service/app/ai/vector_stores/chromadb.py

```diff
@@ -0,0 +1,232 @@
+#  contrary, A) CLOUDERA PROVIDES THIS CODE TO YOU WITHOUT WARRANTIES OF ANY
+#  KIND; (B) CLOUDERA DISCLAIMS ANY AND ALL EXPRESS AND IMPLIED
+#  WARRANTIES WITH RESPECT TO THIS CODE, INCLUDING BUT NOT LIMITED TO
+#  IMPLIED WARRANTIES OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY AND
+#  FITNESS FOR A PARTICULAR PURPOSE; (C) CLOUDERA IS NOT LIABLE TO YOU,
+#  AND WILL NOT DEFEND, INDEMNIFY, NOR HOLD YOU HARMLESS FOR ANY CLAIMS
+#  ARISING FROM OR RELATED TO THE CODE; AND (D)WITH RESPECT TO YOUR EXERCISE
+#  OF ANY RIGHTS GRANTED TO YOU FOR THE CODE, CLOUDERA IS NOT LIABLE FOR ANY
+#  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, PUNITIVE OR
+#  CONSEQUENTIAL DAMAGES INCLUDING, BUT NOT LIMITED TO, DAMAGES
+#  RELATED TO LOST REVENUE, LOST PROFITS, LOSS OF INCOME, LOSS OF
+#  BUSINESS ADVANTAGE OR UNAVAILABILITY, OR LOSS OR CORRUPTION OF
+#  DATA.
+#
+import logging
+from abc import ABC
+from typing import Optional, List, cast, Any, Mapping
+
+import fastapi.exceptions
+from llama_index.core.base.embeddings.base import BaseEmbedding
+from llama_index.core.schema import BaseNode
+from llama_index.core.vector_stores.types import BasePydanticVectorStore
+from llama_index.vector_stores.chroma import (
+    ChromaVectorStore as LlamaIndexChromaVectorStore,
+)
+from llama_index.core.indices import VectorStoreIndex
+import chromadb
+from chromadb.api import ClientAPI
+from chromadb.config import Settings
+from chromadb.api.models.Collection import Collection
+
+from app.ai.vector_stores.vector_store import VectorStore
+from app.config import settings
+from app.services.metadata_apis import data_sources_metadata_api
+from app.services.models import Embedding
+
+logger = logging.getLogger(__name__)
+
+
+def _new_chroma_client() -> ClientAPI:
+    # Note: chromadb.HttpClient requires host and port; ssl, token, database, and tenant are optional depending on server setup
+    # We default to HTTP client to support external ChromaDB servers.
+    try:
+        client_kwargs: dict[str, Any] = {
+            "host": settings.chromadb_host,
+            "headers": (
+                {"X-Chroma-Token": f"{settings.chromadb_token}"}
+                if settings.chromadb_token
+                else None
+            ),
+        }
+
+        if settings.chromadb_database:
+            client_kwargs["database"] = settings.chromadb_database
+        if settings.chromadb_tenant:
+            client_kwargs["tenant"] = settings.chromadb_tenant
+
+        if settings.chromadb_host.startswith("https") and settings.chromadb_server_ssl_cert_path:
+           client_kwargs["settings"] = Settings(chroma_server_ssl_verify=settings.chromadb_server_ssl_cert_path)
+
+        # Only pass port if explicitly provided. If host includes https, Chroma infers SSL.
+        if settings.chromadb_port is not None and not settings.chromadb_host.startswith("https"):
+            client_kwargs["port"] = settings.chromadb_port
```

Comment:
I think we should omit the `https` conditions because

1. The user might specify an SSL-enabled `chromadb_port` without having `"https"` in `chromadb_host`
2. I think?? it's possible to use `https` with ports other than the default inferred `443`, so we might want to still allow specifying port 🤔 

```suggestion
        if settings.chromadb_server_ssl_cert_path:
           client_kwargs["settings"] = Settings(chroma_server_ssl_verify=settings.chromadb_server_ssl_cert_path)

        # Only pass port if explicitly provided. If host includes https, Chroma infers SSL.
        if settings.chromadb_port is not None:
            client_kwargs["port"] = settings.chromadb_port
```